rest/rest-api/src/main/java/org/apache/brooklyn/rest/api/BundleApi.java

```diff
@@ -88,6 +90,51 @@ public BundleSummary detail(
         @PathParam("version")
         String version);
 
+
+    @Path("/{symbolicName}/{version}/types")
+    @GET
+    @ApiOperation(value = "Get all types declared in a given bundle", 
+            response = TypeDetail.class)
+    public List<TypeSummary> getTypes(
+        @ApiParam(name = "symbolicName", value = "Bundle name to query", required = true)
+        @PathParam("symbolicName")
+        String symbolicName,
+        @ApiParam(name = "version", value = "Version of bundle and of type to query", required = true)
+        @PathParam("version")
+        String version);
+    
+    @Path("/{symbolicName}/{version}/types/{typeSymbolicName}")
+    @GET
+    @ApiOperation(value = "Get detail on a given type in a given bundle", 
+            response = TypeDetail.class)
+    public TypeDetail getType(
+        @ApiParam(name = "symbolicName", value = "Bundle name to query", required = true)
+        @PathParam("symbolicName")
+        String symbolicName,
+        @ApiParam(name = "version", value = "Version of bundle and of type to query", required = true)
+        @PathParam("version")
+        String version,
+        @ApiParam(name = "typeSymbolicName", value = "Type name to query", required = true)
+        @PathParam("typeSymbolicName")
+        String typeSymbolicName);
```

Comment:
I'm not sure to understand the difference between this endpoint and the one below (i.e. `/{symbolicName}/{version}/types/{typeSymbolicName}/{typeVersion}`).

It seems to that that it will be the same as `"/{symbolicName}/{version}/types/{typeSymbolicName}/latest` so probably not needed?

Reply:
it isn't `latest` - it is for case where type version = bundle version.  i think as that should be the normal case we should have an endpoint which doesn't require repeating version.

Reply:
Hum I get your PoV. I believe it's better to specify everything but why not, and it's using the same codepath so I'm fine with that